scripts/logid-generator.py

```diff
@@ -65,15 +65,18 @@ def filter_code(_file):
 
   return re.match(r'.*\.(cc|hh)$', _file) != None
 
-def collect_files(directory):
+def collect_files(directory, exclude):
   ''' Collect files from the given directory, filter with filter_code function '''
 
   all_file = []
   for root, dirs, files in os.walk(directory):
-    # remove the unit test directory as we should not check it for log ID-s
-    if "unit" in dirs:
-      del dirs[dirs.index("unit")]
-      continue
+    # remove the excluded directory list as we should not check it for log ID-s
+    if exclude != None:
+      for _dir in exclude:
+        if _dir in dirs:
+          dirs.remove(_dir)
+          exclude.remove(_dir)
```

Comment:
This is not not working as you would expect. The exclude pattern should apply everywhere in the hierarchy, so do not remove it from the excludes list, we want to exclude anything that matches.

E.g.:
aaa/bbb/ccc
aaa/xxx/bbb

exclude pattern bbb  -->  we want to exclude both of the paths above



But it's not general enough. The best solution would be if the user can specify an exclude list where each exclude is a regexp, and that regexp matches on the full file path.

Example:

 -e "a..le" -e "dir_\d+"

Files:
aaa/bbb/ccc.txt
aaa/bbb/apple.html
dir_123/2.txt

The second and third is excluded.